include/llvm/Support/WinTypes.h

```diff
@@ -433,28 +433,69 @@ class CSimpleArray : public std::vector<T> {
 #define IsEqualIID(a, b) a==b
 #define IsEqualCLSID(a, b) !memcmp(&a, &b, sizeof(GUID))
 
-class CW2A {
-public:
-  CW2A(const wchar_t *wide, uint32_t utf = 0) {
-    (void)utf;
-    size_t len = wcslen(wide) + 1; // +1 for '\0'
-    m_psz = (char*) malloc(len * sizeof(char));
-    wcstombs(m_psz, wide, len);
-    m_psz[len-1] = '\0';
-  }
-  ~CW2A() {
-    free(m_psz);
+template< int t_nBufferLength = 128 >
+class CW2AEX
+{
+  public:
+  CW2AEX( LPCWSTR psz , UINT nCodePage = CP_UTF8)
+  {
+    if (nCodePage != CP_UTF8) {
+      // you have reached the limits of this half-assed implementation. So sorry.
```

Comment:
The style guide [requires ](https://google.github.io/styleguide/cppguide.html#Punctuation,_Spelling_and_Grammar)proper punctual comments.
Although a pain at first, I have found this to be nice in the long run.

Can change to:
```cpp
// Current implementation only supports UTF8.
```

Reply:
hah! sorry. I forgot about that comment. It's the kind of snark I'd include in reviews but not source code.